### DISCUSSION TITLE
Is there a reason for the lack of "case" dependency in spacy.symbol?

### DISCUSSION THREAD

**Discussion by jesusfbes** [explosion/spaCy#8583] [Help: Other Questions] [answered]
Hi to all,

We recently found out that there is not a symbol in "spacy.symbol" for some dependencies. For instance "case" dependency. We would like to do operations like the following with case:

```from spacy.symbols import nsubj

for t in doc:
          if t.dep == nsubj:                        # dependency 'nsubj' IS recognized
                 print(f"Found nsub token")
```
        
Is there a reason for this? Is there a workaround to solve this in a clean way without using the string and "dep_" attribute?

**Comment by qingyun1988** [comment]
There is no 'case' in spacy, it is replaced by 'prep' and 'poss'.

**Reply by adrianeboyd** [reply]
`case` is a valid dependency label, but this is a separate question about the symbols in `spacy.symbols`.

**Comment by adrianeboyd** [comment, marked as answer]
This is more for historical reasons than anything, and whether a string is in `spacy.symbols` only matters for the symbols in `spacy.attrs` at this point. We haven't removed any symbols to maintain backwards compatibility, but with the string hashes introduced in spacy v2, many of these symbols aren't really needed anymore.

You should always get the same integer from `nlp.vocab.strings["case"]` for the label you're interested in, so I wouldn't use `spacy.symbols` for this. If you want to be sure that you're using valid labels, you can refer to the model's labels (`nlp.get_pipe("parser").labels`) and convert them to integers with `nlp.vocab.strings[label]`. All internal integer representations of these strings will correspond to the value you get from the `StringStore` in `nlp.vocab.strings`. The same hash function is used across all models, so no matter which pipeline you've loaded, `nlp.vocab.strings[label]` should always return the same value.

**Reply by jesusfbes** [reply]
Very clear @adrianeboyd.

Thank you very much for your response.
